samcli/local/apigw/local_apigw_service.py

```diff
@@ -220,6 +220,12 @@ def _parse_lambda_output(lambda_output, binary_types, flask_request):
             LOG.error(message)
             raise TypeError(message)
 
+        # API Gateway only accepts statusCode, body, headers, and isBase64Encoded in
+        # a response shape.
+        if LocalApigwService._invalid_apig_response(json_output):
+            msg = "Invalid Response: " + str(json_output)
```

Comment:
do we need to be more explicit in the response about accepted keys?

Reply:
we can possibly move log statement up here just before throwing an exception.

Reply:
To do this I'd probably want to actually return the invalid keys and then perform the boolean if at this top level. Should we move to that approach?

Reply:
Yeah that sounds good, provides more context as well.

Reply:
Addressed in most recent change.